sdks/python/apache_beam/transforms/trigger.py

```diff
@@ -633,11 +683,14 @@ def to_runner_api(self, context):
 
 class TriggerContext(object):
 
-  def __init__(self, outer, window):
+  def __init__(self, outer, window, clock):
     self._outer = outer
     self._window = window
+    self._clock = clock
 
   def set_timer(self, name, time_domain, timestamp):
+    if self._clock and time_domain == TimeDomain.REAL_TIME:
+      timestamp += self._clock.time()
     self._outer.set_timer(self._window, name, time_domain, timestamp)
```

Comment:
<!--new_thread; commit:a769afcabd2a4b8ad94cae1f1c62e73823e42d7b; resolved:0-->
It seems like maybe we should do this `+= self._clock.time()` in the base `TriggerDriver`.  It seems odd to have this inconsistency where in one `set_timer` call, we use an offset from the current time, and in another, we use an absolute time.